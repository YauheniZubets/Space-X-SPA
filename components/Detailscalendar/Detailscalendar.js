import React from 'react';
import PropTypes from 'prop-types';
import Main from '../Main/Main';
import {Link} from 'react-router-dom';

import Preloadcircle from '../Preloadcircle/Preloadcircle';
import Preloadsquare from '../Preloadsquare/Preloadsquare';

import './Detailscalendar.css';

class Detailscalendar extends React.PureComponent {

    static propTypes = {
        currentLaunch: PropTypes.object,
        pagPage:PropTypes.number //номер страницы для возврата на компонент Календарь
    }

    state = {
        isVideoLoaded: false // проверяем загрузилось ли видео, если загружено прелоадер удаляется
    }

    componentWillReceiveProps() { // сброс состояния прелоадера для видео
        this.setState({isVideoLoaded: false})
    }

    render () {
        console.log(this.props);
        const launch=this.props.currentLaunch;
        return (
            <div>
                {
                    (!this.props.currentLaunch) ?
                    <div style={{height: '100vh'}}><Preloadsquare /></div> :
                    <div>
                        <Main name={launch.name} />
                        <div className="details">
                            <div className="container">
                                <div className="details-row">
                                    <div className="details-image">
                                        <img src={launch.links.patch.small} alt={launch.name} />
                                    </div>
                                    <div className="details-content">
                                        <p className="details-description">{launch.details}</p>
                                    </div>
                                </div>
                                <Preloadcircle hide={this.state.isVideoLoaded} />
                                <iframe 
                                    className={this.state.isVideoLoaded ? 'details-youtube' : 'video-display'} 
                                    width="560" 
                                    height="315" 
                                    src={`https://www.youtube.com/embed/${launch.links.youtube_id}`} 
                                    frameBorder="0" 
                                    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" 
                                    allowFullScreen
                                    onLoad={()=>this.setState({isVideoLoaded: true})}>
                                </iframe>
                            </div>
                            <Link to={`/starts/page/${this.props.pagPage}`} className="button button-back">go back</Link>
                        </div>
                    </div>
                }
            </div>
        )
    }
};

export default Detailscalendar;