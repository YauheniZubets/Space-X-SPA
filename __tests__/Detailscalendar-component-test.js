import React from 'react';
import renderer from 'react-test-renderer';

import Detailscalendar from '../components/Detailscalendar/Detailscalendar';

test('работа DetailscalendarComponent', () => {

  const detailsCalendar = renderer.create(
    <Detailscalendar />
  );

  let componentTree=detailsCalendar.toJSON();
  
  expect(componentTree).toMatchSnapshot();
  
  expect(componentTree).toMatchSnapshot();
    
});
