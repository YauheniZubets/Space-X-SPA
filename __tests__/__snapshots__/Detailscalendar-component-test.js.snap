// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа DetailscalendarComponent 1`] = `
<div>
  <div
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="loader"
    >
      <div
        className="l_main"
      >
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`работа DetailscalendarComponent 2`] = `
<div>
  <div
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <div
      className="loader"
    >
      <div
        className="l_main"
      >
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          className="l_square"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
